legend-engine-language-pure-dsl-persistence/src/main/java/org/finos/legend/engine/language/pure/dsl/persistence/compiler/toPureGraph/HelperPersistenceBuilder.java

```diff
@@ -0,0 +1,482 @@
+        private final CompileContext context;
+
+        private PersisterBuilder(CompileContext context)
+        {
+            this.context = context;
+        }
+
+        @Override
+        public Root_meta_pure_persistence_metamodel_Persister visit(BatchPersister val)
+        {
+            return new Root_meta_pure_persistence_metamodel_batch_BatchPersister_Impl("")
+                    ._targetSpecification(buildTargetSpecification(val.targetSpecification, context));
+        }
+
+        @Override
+        public Root_meta_pure_persistence_metamodel_Persister visit(StreamingPersister val)
+        {
+            return new Root_meta_pure_persistence_metamodel_streaming_StreamingPersister_Impl("");
+        }
+    }
+
+    private static class TargetSpecificationBuilder implements TargetSpecificationVisitor<Root_meta_pure_persistence_metamodel_batch_targetspecification_TargetSpecification>
+    {
+        private final String targetName;
+        private final Class<?> modelClass;
+        private final CompileContext context;
+
+        private TargetSpecificationBuilder(String targetName, Class<?> modelClass, CompileContext context)
+        {
+            this.targetName = targetName;
+            this.modelClass = modelClass;
+            this.context = context;
+        }
+
+        @Override
+        public Root_meta_pure_persistence_metamodel_batch_targetspecification_TargetSpecification visit(FlatTargetSpecification val)
+        {
+            return buildFlatTargetSpecification(val, modelClass, context);
+        }
+
+        @Override
+        public Root_meta_pure_persistence_metamodel_batch_targetspecification_TargetSpecification visit(GroupedFlatTargetSpecification val)
+        {
+            return new Root_meta_pure_persistence_metamodel_batch_targetspecification_GroupedFlatTargetSpecification_Impl("")
+                    ._modelClass(modelClass)
+                    ._transactionScope(context.resolveEnumValue(PERSIST_PACKAGE_PREFIX + "::batch::targetspecification::TransactionScope", val.transactionScope.name()))
+                    ._components(ListIterate.collect(val.components, c -> resolveComponent(c, modelClass, context)));
+        }
+
+        @Override
+        public Root_meta_pure_persistence_metamodel_batch_targetspecification_TargetSpecification visit(NestedTargetSpecification val)
+        {
+            return new Root_meta_pure_persistence_metamodel_batch_targetspecification_NestedTargetSpecification_Impl("")
+                    ._targetName(targetName)
+                    ._modelClass(modelClass);
+        }
+
+        private Root_meta_pure_persistence_metamodel_batch_targetspecification_FlatTargetSpecification buildFlatTargetSpecification(FlatTargetSpecification specification, Class<?> modelClass, CompileContext context)
+        {
+            return new Root_meta_pure_persistence_metamodel_batch_targetspecification_FlatTargetSpecification_Impl("")
+                    ._targetName(targetName)
```

Comment:
This is always null when we reach here thru GroupedFlatTargetSpecification.  The name is assigned from the top level (GroupedFlatTargetSpecification), which does not really have a name.

I suggest to remove the targetName field, and always read the name from the spec been visited.

Thoughts?

Reply:
For other backends, we need a way to add add'l configuration to individual datasets in grouped flat.

We could do it through property name instead ...

Are your determining dataset name from the name of the target class (or tagged value on that name)?

Reply:
At the moment I'm using the name.  I could switch to property on PropertyAndFlatTargetSpecification, but not sure is correct.  The reason I pick it from the targetName is because of naming conventions - pure property naming would limit the name of a table/dataset name.

Reply:
Are the targetName refactoring something you currently working on?  

The current pure model I followed (https://github.com/finos/legend-pure/pull/442) does not define targetName at TargetSpecification either.  

We might want to outline the pure changes you are planning to see how I can incorporate them incrementally without breaking my use case.

Reply:
Ah sorry -- I thought you were requesting this in your initial comment ...

I'll look into why it's null

Reply:
Should be fixed.

Reply:
Awesome - thanks!

Reply:
BTW - Should we move this out of draft state?

Reply:
Done. And agree on enumerating upcoming Pure changes, will put together the list.